opentech/apply/determinations/tests/test_views.py

```diff
@@ -0,0 +1,104 @@
+from django.test import TestCase, RequestFactory
+from django.urls import reverse
+
+from opentech.apply.users.tests.factories import StaffFactory, UserFactory
+from .factories import DeterminationFactory
+from opentech.apply.funds.tests.factories import ApplicationSubmissionFactory
+
+
+class BaseTestCase(TestCase):
+    url_name = ''
+    user_factory = None
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = self.user_factory()
+        self.client.force_login(self.user)
+
+    def url(self, instance, view_name='detail'):
+        full_url_name = self.url_name.format(view_name)
+        url = reverse(full_url_name, kwargs=self.get_kwargs(instance))
+        request = self.factory.get(url, secure=True)
+        return request.build_absolute_uri()
+
+    def get_page(self, instance, view_name='detail'):
+        return self.client.get(self.url(instance, view_name), secure=True, follow=True)
+
+    def post_page(self, instance, data, view_name='detail'):
+        return self.client.post(self.url(instance, view_name), data, secure=True, follow=True)
+
+    def refresh(self, instance):
+        return instance.__class__.objects.get(id=instance.id)
+
+
+class StaffDeterminationsTestCase(BaseTestCase):
+    user_factory = StaffFactory
+    url_name = 'funds:submissions:determinations:{}'
+
+    def get_kwargs(self, instance):
+        return {'submission_pk': instance.submission.id}
+
+    def test_can_access_determination(self):
+        submission = ApplicationSubmissionFactory(status='in_discussion')
+        determination = DeterminationFactory(submission=submission, author=self.user, not_draft=True)
+        response = self.get_page(determination)
+        self.assertContains(response, determination.submission.title)
+        self.assertContains(response, self.user.full_name)
+        self.assertContains(response, reverse('funds:submissions:detail', kwargs={'pk': submission.id}))
```

Comment:
Should this have a not contains for the extra data?

Reply:
plus lead should see data

Reply:
sorted